vector/src/main/java/im/vector/app/features/home/room/detail/timeline/action/MessageActionsViewModel.kt

```diff
@@ -333,6 +334,13 @@ class MessageActionsViewModel @AssistedInject constructor(
     ) {
         val eventId = timelineEvent.eventId
         if (!timelineEvent.root.isRedacted()) {
+            if (initialState.isFromPinnedMessagesTimeline) {
+                if (actionPermissions.canPinMessage && vectorPreferences.arePinnedMessagesEnabled()) {
+                    add(EventSharedAction.UnpinMessage(eventId))
+                    add(EventSharedAction.ViewPinnedMessageInRoom(eventId))
+                }
+                return
```

Comment:
I understand why there is a `return` here, because there is no other action when in this screen. But I think it can lead to errors.
Maybe add a `else` block instead?